reconcile/utils/terraform_client.py

```diff
@@ -531,34 +531,62 @@ def cleanup(self):
         should be skipped.
         """
+        # A commonly changed RDS database settings that can be applied either
+        # immediately or during the next scheduled maintenance window, see:
+        #  https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Overview.DBInstance.Modifying.html#USER_ModifyInstance.Settings
+        #
+        # Note that changing the RDS database identifier (the database name) is
+        # a complex operation that also involves Terraform performing a resource
+        # replacement and thus it's not included here, see:
+        #   https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_RenameInstance.html
+        supported_values: dict[str, str] = {
+            "engine_version": "EngineVersion",
+            "storage_type": "StorageType",
+            "allocated_storage": "AllocatedStorage",
+            "instance_class": "DBInstanceClass",
+            "multi_az": "MultiAZ",
+            "iops": "Iops",
+        }
         before = resource_change["before"]
         after = resource_change["after"]
-        changed_terraform_args = [
+        changed_resource_arguments = [
             key for key, value in before.items() if value != after.get(key)
         ]
-        if (
-            len(changed_terraform_args) == 1
-            and "engine_version" in changed_terraform_args
-            and self._aws_api is not None
-        ):
+        if changed_resource_arguments and self._aws_api is not None:
             region_name = get_region_from_availability_zone(before["availability_zone"])
             response = self._aws_api.describe_rds_db_instance(
                 account_name, resource_name, region_name=region_name
             )
-            pending_modified_values = response["DBInstances"][0][
-                "PendingModifiedValues"
-            ]
-            if (
-                "EngineVersion" in pending_modified_values
-                and pending_modified_values["EngineVersion"]
-                == resource_change["after"]["engine_version"]
-            ):
-                return True
-        return False
+            try:
+                pending_modified_values = response["DBInstances"][0].get(
+                    "PendingModifiedValues"
+                )
+            except (KeyError, IndexError):
+                raise RuntimeError(f"Expected RDS database {resource_name} to exist")
+            if pending_modified_values is None:
+                # The PendingModifiedValues attribute might not be included as
+                # part of the RDS database instance object. However, we expect
+                # it to be included here, especially since, at this point,
+                # Terraform also claims a pending change for the underlying
+                # resource. Thus if we can't agree on AWS API vs Terraform
+                # state, it's better to raise to signify a potential problem.
+                raise RuntimeError(
```

Comment:
**To the reviewer:**

The chances of this occurring are minuscule, albeit non-zero. However, if this does happen, it would indicate that there is something wrong with the state Terraform currently holds and the future changes that are planned to be executed.

Reply:
I will change this to just emit a log line at a debug level as there might be cases where we run and the change had just been completed.